### PR TITLE
chore: integrate existing contract code and clean up

### PR DESCRIPTION
**Pull request by karlwuest**
- Integrates most of the existing code from `blob_store` in the new package (excluding epoch change functions)
- Removes generic coin type and replaces it with SUI
- Cleans up some of the code (restrict access for previously `public` functions, change to method syntax)
- Re-activates code coverage for move (walrus package)
- Doesn't include any functional changes

Next steps:
- Refactor certified messages and integrate `invalidate_blob_id` again
- Implement Blob life cycle changes (deletion, extension, write payments)
- Simplify committee and BLS verification code
- Implement system-side epoch change logic

### CODE REVIEW

File: contracts/walrus/sources/system/blob.move [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+    /// After the period of validity expires for the blob we can destroy the blob resource.
+    public fun destroy_blob_object(sys: &System, blob: Blob) {
+        let current_epoch = sys.epoch();
+        assert!(current_epoch >= blob.storage.end_epoch(), EResourceBounds);
+
+        // Destroy the blob
+        let Blob {
+            id,
+            stored_epoch: _,
```
Comment (damirka) [inline contracts/walrus/sources/system/blob.move]: You can use the `..` operator

File: contracts/walrus/sources/system/bls_aggregate.move [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        /// The total number of shards held by the committee
+        n_shards: u16,
+    }
+
+    /// Constructor
+    public fun new_bls_committee(members: vector<StorageNodeInfo>): BlsCommittee {
+        // Compute the total number of shards
+        let mut n_shards = 0;
+        let mut i = 0;
+        while (i < members.length()) {
```
Comment (damirka) [inline contracts/walrus/sources/system/bls_aggregate.move]: If you want to optimize:
```
members.length().do!(|x| {
 // .. while body
}):
```

File: contracts/walrus/sources/system/committee.move [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        epoch: u64,
+        members: vector<StorageNodeInfo>,
+    ): Committee {
+        // Make BlsCommittee
+        let bls_committee = new_bls_committee(members);
+
+        Committee { epoch, bls_committee }
+    }
+
+    #[test_only]
```
Comment (damirka) [inline contracts/walrus/sources/system/committee.move]: It's better to put `#[test_only]` in the bottom always. As a reviewer as soon as I see this I don't expect anything important after it.

File: contracts/walrus/sources/system/redstuff.move [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        min_u16(max_byzantine(n_shards) / 5, 5)
+    }
+
+    /// Maximum number of byzantine shards, given `n_shards`.
+    fun max_byzantine(n_shards: u16): u16 {
+        (n_shards - 1) / 3
+    }
+
+    fun min_u16(a: u16, b: u16): u16 {
+        if (a < b) {
```
Comment (damirka) [inline contracts/walrus/sources/system/redstuff.move]: Function seems redundant, we have `a.min(b)`

File: contracts/walrus/sources/system/storage_accounting.move [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        current_index: u64,
+        length: u64,
+        ring_buffer: vector<FutureAccounting>,
+    }
+
+    /// Constructor for FutureAccountingRingBuffer
+    public(package) fun ring_new(length: u64): FutureAccountingRingBuffer {
+        let mut ring_buffer: vector<FutureAccounting> = vector::empty();
+        let mut i = 0;
+        while (i < length) {
```
Comment (damirka) [inline contracts/walrus/sources/system/storage_accounting.move]:
```move
let right_buffer = vector::tabulate!(length, |epoch| FutureAccounting { 
   epoch,
   storage_to_reclaim: 0,
   rewards_to_distribute: balance::zero()
});
```

File: contracts/walrus/sources/system/redstuff.move (hunk 4 above)
Comment (karlwuest) [reply contracts/walrus/sources/system/redstuff.move]: Ah nice, that's a remnant from when (as far as I know) that didn't exist

**Pull request merged**